shared-module/usb_hid/Device.h

```diff
@@ -54,4 +54,10 @@ extern const usb_hid_device_obj_t usb_hid_device_consumer_control_obj;
 
 void usb_hid_device_create_report_buffers(usb_hid_device_obj_t *self);
 
+typedef struct {
+    char interface_name[128];
+} usb_hid_interface_name_t;
+
+extern usb_hid_interface_name_t usb_hid_interface_name_obj;
```

Comment:
Instead of reserving 128 bytes of RAM all the time for this, use the technique used in `supervisor.set_usb_identification()`, https://github.com/adafruit/circuitpython/blob/54e78c89fc9e0d2b31544dfd03c30a159e8540a4/shared-bindings/supervisor/__init__.c#L297-L298. It allocates space for the user-set string(s) only if they are specified. It uses `port_malloc()`, which allocates outside the Python heap (so no gc issues). `port_malloc()` is new in 9.0.0.